docs/tutorial.rst

```diff
@@ -100,7 +100,69 @@ large sections of the tree which do not have any variable sites using
 
 .. _troubleshooting:
 
+*******************************************************
+Inferring and Dating Tree Sequences with Ancient Samples
+*******************************************************
 
+``tsdate`` and ``tsinfer`` can be used together to infer tree sequences from both
+modern and ancient samples. The following recipe shows how this is accomplished
+with a few lines of Python.
+
+.. code-block:: python
+
+   import msprime
+   import tsdate
+   import tsinfer
+   import tskit
+
+   import numpy as np
+
+
+   def historical_samples_example():
+       samples = [
+           msprime.Sample(population=0, time=0),
+           msprime.Sample(0, 0),
+           msprime.Sample(0, 0),
+           msprime.Sample(0, 0),
+           msprime.Sample(0, 1.0),
+           msprime.Sample(0, 1.0)
+       ]
+       return msprime.simulate(samples=samples, mutation_rate=1, length=100)
+
+   ts = historical_samples_example()  # Simulate tree sequence
+   samples = tsinfer.SampleData.from_tree_sequence(
+      ts, use_sites_time=False) # Get SampleData file
+   modern_samples = samples.subset(np.where(samples.individuals_time[:] == 0)[0])
+   # The following line removes mutations which are only present in ancient samples
```

Comment:
We don't really need to do this if we change the simplify call to simplify(filter_sites=False)

Reply:
Ah, so it's only to make sure the site IDs are unchanged? I think it would be better to put this in the `simplify` call, which makes it explicit (and also saves a line of code).

Reply:
It removes any "ancient-only" mutations so that all the sites will line up when calling `np.maximum()` in `add_sampledata_times()`. But this isn't necessary if we ensure that all the ancient-only sites are also present in the tree sequence inferred from only modern samples. The only thing is that this won't be the case with real data I expect...

Reply:
But actually if we tell them to come with a sampledata file of moderns and ancients, then it'll work out fine with this example.